### ISSUE TITLE
Not able to search using doSearch Method of  https://gw.sam.gov/SAMWS/1.0/ExclusionSearch?wsdl

### ISSUE THREAD

**Issue by yacu01** [GSA/sam_api#58]
I am having trouble using the doSearch operation of https://gw.sam.gov/SAMWS/1.0/ExclusionSearch?wsdl.   It just does not return anything even when I tried to use the partial name as the only input. I am using soapui and i have attached the request along with the results. Can you direct me as to what I am doing wrong here?

![sam_png](https://cloud.githubusercontent.com/assets/1162748/5725194/cfb58abe-9b1e-11e4-88d4-1e7d5fec9711.png)

**Comment by pammiller0** [comment]
Hi - sorry for the late reply. We have the team looking at this. We will get back to you very soon.

**Comment by Rafael-IBM** [comment]
@yacu01 can you manually go into the XML and delete the fields except partialName?  That's what I've found works best.

![untitled](https://cloud.githubusercontent.com/assets/10468868/5941857/aa4fad9e-a6e1-11e4-84e0-467f50f440e2.png)

**Comment by mrgrita** [comment]
I am having the same problem where I cannot get any successful results when using doSearch operation. I am using .Net 4.0 and your ExclusionSearchServiceBeanClient. The doSearch operation expects a value of type "OperationSearchType" and not an xml string, so I cannot edit the SOAP message and send that as a value for the doSearch method.

Below is an example of C# code I am using:
ExclusionSearchServiceBeanClient proxy = new ExclusionSearchServiceBeanClient();
ExclusionSearchResponseType response = new ExclusionSearchResponseType();
OperationSearchType os = new OperationSearchType();
os.name = "CompanyName";
response = proxy.doSearch(os);

I was using similar code to the above before the new web service was implemented in November of 2014 and it was working. Now, I just get message: "You may retrieve a maximum of 1000 records per XML request. The search criteria that you specified in your request match more records than this threshold; please refine your search criteria to narrow down the scope of your search and try again"

**Comment by pammiller0** [comment]
@Rafael-IBM hey Rafael - Can you look into this issue? Pam

**Comment by Rafael-IBM** [comment]
This is a response from @mfedorov on how he resolved the doSearch with .NET.  See below:

@Rafael-IBM, the response message from doSearch contains empty dates, causing SoapUI to fail. For example, <TermDt/>. This field is not marked as nillable but comes in response without any value. I modified date parser on my side to get it working programmatically with Apache CXF. 

So this is an issue (fields being nillable) that is one we are looking at.  It seems to be an issue with .NET.  The above is how one developer got around it.

**Comment by pammiller0** [comment]
@yacu01 @mrgrita - have you been able to resolve the issue with .net? As mentioned above another developer modified date parser and was able to get it working. - Let me know....

**Comment by mrgrita** [comment]
No, I have not been able to get a successful reply from the web service. I am working in .Net 4.0 framework and the request it generates is not accepted by the SAM web service. I am currently trying to edit the request.

**Comment by mrgrita** [comment]
Update: I am now able to receive successful responses from the SAM web service. I added a class to modify the request to the web service with help from the following blog: http://blogs.msdn.com/b/stcheng/archive/2009/02/21/wcf-how-to-inspect-and-modify-wcf-message-via-custom-messageinspector.aspx.

I first had to revise the web service's .Net generated reference.cs file to add attribute:'Use=System.ServiceModel.OperationFormatUse.Encoded' for System.ServiceModel.XmlSerializerFormatAttribute tag in the doSearch and doSsnSearch methods. I then created my MessageInspector class as referenced in the blog. Using the MessageInspector class, I retrieved the fields (child elements) from the 'OperationSearchType' element and added them as child elements of the 'exclusionSearchCriteria' element.

Request before edit:

<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
   <s:Header>
     <Action s:mustUnderstand="1" xmlns="http://schemas.microsoft.com/ws/2005/05/addressing/none" />
   </s:Header>
   <s:Body s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
      <q1:doSearch xmlns:q1="http://www.sam.gov">
         <exclusionSearchCriteria href="#id1" />
      </q1:doSearch>
      <q2:OperationSearchType id="id1" xsi:type="q2:OperationSearchType" xmlns:q2="http://www.sam.gov/encoded">
         <includeArchive xsi:type="xsd:string">N</includeArchive>
         <name xsi:type="xsd:string">searchEntityName</name>
      </q2:OperationSearchType>
   </s:Body>
</s:Envelope>

Request after edit:

<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
   <s:Header>
      <Action s:mustUnderstand="1" xmlns="http://schemas.microsoft.com/ws/2005/05/addressing/none" />
   </s:Header>
   <s:Body s:encodingStyle="http://schemas.xmlsoap.org/soap/encoding/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
      <q1:doSearch xmlns:q1="http://www.sam.gov">
         <exclusionSearchCriteria href="#id1">
            <includeArchive xsi:type="xsd:string">N</includeArchive>
            <name xsi:type="xsd:string">searchEntityName</name>
         </exclusionSearchCriteria>
      </q1:doSearch>
      <q2:OperationSearchType id="id1" xsi:type="q2:OperationSearchType" xmlns:q2="http://www.sam.gov/encoded"></q2:OperationSearchType>
   </s:Body>
</s:Envelope>

**Comment by pammiller0** [comment]
@mrgrita Thanks for posting your fix for all to see! @yacu01 have you had any luck?
